ROX-27622: Move SecurityPolicy CRD to template dir in Helm chart (#13781)
stackrox/stackrox: CHANGELOG.md

## [NEXT RELEASE]

> **HELM USERS**: Please see the ROX-27622 under "technical changes" to avoid upgrade failures!

Reviewer: ```suggestion
**HELM USERS**: Please see ROX-27622 under "technical changes" to avoid upgrade failures!
```
Not a native speaker, feel free to ignore.

- ROX-28263: New `roxctl` help formatting.
- ROX-24500: Certificate validation failure in `roxctl` is now an error.
- ROX-27622: Move `SecurityPolicy` CRD to template directory in Helm chart. **All Helm users will need to take action!**
  No action is needed for users that use the operator or `roxctl` to install StackRox.
  This change makes the CRD simpler to maintain for users because it will now be automatically upgraded.

>   To avoid installation failure, Helm users need to apply the following changes to the CRD prior to upgrade:

Reviewer 2: ```suggestion
  To avoid upgrade failure, Helm users need to apply the following changes to the CRD prior to upgrade:
```

      kubectl annotate crd/securitypolicies.config.stackrox.io meta.helm.sh/release-name=stackrox-central-services
      kubectl annotate crd/securitypolicies.config.stackrox.io meta.helm.sh/release-namespace=stackrox
      kubectl label crd/securitypolicies.config.stackrox.io app.kubernetes.io/managed-by=Helm

>   The value of the annotations will need to be updated if you've updated the release name (i.e. "stackrox-central-services") or namespace (i.e. "stackrox").

Reviewer 2: This sentence could be misunderstood, given that the commands update annotation values. How about:

```suggestion
  The above values will need to be updated to match your release name (i.e. "stackrox-central-services") or namespace (i.e. "stackrox") in case you had used different ones.
```

### Technical Changes

- Scanner V4 now uses [Red Hat's VEX files](https://security.access.redhat.com/data/csaf/v2/vex/) instead of the [CVE map](https://security.access.redhat.com/data/metrics/cvemap.xml) for vulnerability data related to non-RPM content inside of official Red Hat images.
- ROX-27622: Move `SecurityPolicy` CRD to template directory in Helm chart. **All Helm users will need to take action!**
  No action is needed for users that use the operator or `roxctl` to install StackRox.
  This change makes the CRD simpler to maintain for users because it will now be automatically upgraded.
  To avoid installation failure, Helm users need to apply the following changes to the CRD prior to upgrade:

      kubectl annotate crd/securitypolicies.config.stackrox.io meta.helm.sh/release-name=stackrox-central-services

>       kubectl annotate crd/securitypolicies.config.stackrox.io meta.helm.sh/release-namespace=stackrox

Reviewer 3: We do allow configuring the namespace to other than stackrox right? So this line will have to accommodate that.
